extensions/kubernetes-client/runtime-internal/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesSerializationRecorder.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.kubernetes.client.runtime;
+
+import java.util.HashSet;
+import java.util.ServiceLoader;
+import java.util.Set;
+
+import io.fabric8.kubernetes.api.model.KubernetesResource;
+import io.fabric8.kubernetes.internal.KubernetesDeserializer;
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class KubernetesSerializationRecorder {
+
+    public RuntimeValue<Class<? extends KubernetesResource>[]> scanKubernetesResources() {
+        final var classLoaders = new ClassLoader[] {
+                Thread.currentThread().getContextClassLoader(),
+                KubernetesDeserializer.class.getClassLoader() };
+        final Set<Class<? extends KubernetesResource>> resourceClasses = new HashSet<>();
+        for (var classLoader : classLoaders) {
+            if (classLoader == null) {
+                continue;
+            }
+            final var serviceLoader = ServiceLoader.load(KubernetesResource.class, classLoader);
+            for (var kr : serviceLoader) {
+                resourceClasses.add(kr.getClass());
+            }
+        }
+        return new RuntimeValue<>(resourceClasses.<Class<? extends KubernetesResource>> toArray(Class[]::new));
+    }
```

Comment:
This can be done at build time can't it?

Reply:
I mean the scanning can be done at build time and the result can then be "passed" to the recorder.

Reply:
Let me try to split it into two parts, but I think that the key here is accessing these classes during the static init phase (it's all about working around the SPI issue).
Anyway, I'll let you know how it goes.

Reply:
Yes, that is the key, but my point is that the discovery part can be done at build time and just have the recorder load the classes so GraalVM takes then into account